Update contributor docs to include more details on running integration tests. (#2632)
GoogleContainerTools/jib: CONTRIBUTING.md

5. Associate the change with an existing issue or file a [new issue](../../issues).
6. Create a pull request!

### Integration Tests
**Note** that in order to run integration tests, you will need to set one of the
following environment variables:                                                

  - If you are using a GCP project then set `JIB_INTEGRATION_TESTING_PROJECT` to the GCP project to use for testing;
    the registry tested will be `gcr.io/<JIB_INTEGRATION_TESTING_PROJECT>`.
    - Configure authentication to Container Registry by following these [steps](https://cloud.google.com/container-registry/docs/advanced-authentication).

>     - Enable the Google Container Registry api [here](https://pantheon.corp.google.com/apis/library/containerregistry.googleapis.com).

Reviewer: ```suggestion
    - Enable the Google Container Registry API [here](https://console.cloud.google.com/apis/library/containerregistry.googleapis.com).
```

5. Associate the change with an existing issue or file a [new issue](../../issues).
6. Create a pull request!

### Integration Tests
**Note** that in order to run integration tests, you will need to set one of the
following environment variables:                                                

  - If you are using a GCP project then set `JIB_INTEGRATION_TESTING_PROJECT` to the GCP project to use for testing;
    the registry tested will be `gcr.io/<JIB_INTEGRATION_TESTING_PROJECT>`.
    - Make sure that you have `docker-credential-gcr` installed.

>     - Configure authentication to Container Registry by following these [steps](https://cloud.google.com/container-registry/docs/advanced-authentication).

Reviewer: I think actually these two are conflicting, according to the linked doc. The doc recommends the first option of ["gcloud as a credential helper"](https://cloud.google.com/container-registry/docs/advanced-authentication#gcloud-helper). `docker-credential-gcr` is the second ["standalone credential helper"](https://cloud.google.com/container-registry/docs/advanced-authentication#standalone-helper) option.

Author: Great point, thanks! I removed the installation step since it was conflicting with the one after.

>     - Enable the google container registry api [here](https://pantheon.corp.google.com/apis/library/containerregistry.googleapis.com). This can also be done through the command line.

Reviewer: ```suggestion
    - Enable the Google Container Registry API [here](https://console.cloud.google.com/apis/library/containerregistry.googleapis.com).
```
And unless you document the sample command, I'd skip mentioning the command-line opiton.

>   - If you're not using a GCP project then set `JIB_INTEGRATION_TESTING_LOCATION` to a specific registry for testing.        
> 
> To run the integration tests locally, you can run `docker run -d -p 9990:5000 registry:2` and use `localhost:9990`.

Reviewer: How about this? The `localhost:9990` is for `JIB_INTEGRATION_TESTING_LOCATION` actually.

```suggestion
  - If you're not using a GCP project then set `JIB_INTEGRATION_TESTING_LOCATION` to a specific registry for testing. (For example, you can run `docker run -d -p 9990:5000 registry:2` to set up a local registry and set the variable to `localhost:9990`.)
```

Author: Ohh I see! Done.
